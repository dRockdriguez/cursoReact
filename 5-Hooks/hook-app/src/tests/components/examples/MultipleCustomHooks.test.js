import React from "react";
import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
import { useCounter } from "../../../hooks/useCounter";

jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("MultipleCustomHooks", () => {
  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 1,
      increment: () => {},
    });
  });
  
  test("Debe mostrarse correctamente", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: null,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe  mostrar la informcion", () => {
    useFetch.mockReturnValue({
      data: [{ author: "David", quote: "Hola mundo" }],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".blockquote p").text().trim()).toBe("Hola mundo");
    expect(wrapper.find(".blockquote-footer").text().trim()).toBe("David");
  });
});
