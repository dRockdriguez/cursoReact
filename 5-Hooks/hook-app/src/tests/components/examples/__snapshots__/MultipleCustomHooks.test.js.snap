// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleCustomHooks Debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    Breaking bad
  </h1>
  <hr />
  <blockquote
    className="blockquote text-end"
  >
    <p />
    <footer
      className="blockquote-footer"
    />
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</Fragment>
`;
